ninja-core/src/main/java/ninja/validation/Validators.java

```diff
@@ -16,40 +16,89 @@
+    private static String fieldKey(String fieldName, String configuredFieldKey) {
+        if (configuredFieldKey.length() > 0) {
+            return configuredFieldKey;
+        } else {
+            return fieldName;
+        }
+    }
+
+    /**
+     * A basic Message interpolator.
+     *
+     * @author Thibault Meyer
+     */
+    private static class NinjaContextMsgInterpolator implements MessageInterpolator.Context, Serializable {
+
+        private final Object value;
+        private final ConstraintDescriptor<?> descriptor;
+
+        /**
+         * Create message interpolator context.
+         *
+         * @param value      value being validated
+         * @param descriptor Constraint being validated
+         */
+        public NinjaContextMsgInterpolator(Object value, ConstraintDescriptor<?> descriptor) {
+            this.value = value;
+            this.descriptor = descriptor;
+        }
 
+        @Override
+        public ConstraintDescriptor<?> getConstraintDescriptor() {
+            return descriptor;
+        }
 
         @Override
+        public Object getValidatedValue() {
+            return value;
+        }
+    }
+
+    public static class JSRValidator implements Validator<Object> {
+
+        @Override
+        @Deprecated
         public void validate(Object value, String field, Validation validation) {
-            if (value == null) {
-                return;
-            }
-            ValidatorFactory validatorFactory = javax.validation.Validation
-                    .buildDefaultValidatorFactory();
-            javax.validation.Validator validator = validatorFactory
-                    .getValidator();
-            Set<javax.validation.ConstraintViolation<Object>> violations = validator
-                    .validate(value);
-
-            for (javax.validation.ConstraintViolation<Object> violation : violations) {
-                ConstraintViolation constraintViolation =
-                        ConstraintViolation.create(violation.getMessage(), violation
-                                .getInvalidValue());
-
-                validation.addBeanViolation(new FieldViolation(violation
-                        .getPropertyPath().toString(), constraintViolation));
+            this.validate(value, field, validation, null);
+        }
+
+        @Override
+        public void validate(Object value, String field, Validation validation, ninja.Context context) {
+            if (value != null) {
+                final ValidatorFactory validatorFactory = javax.validation.Validation.buildDefaultValidatorFactory();
+                final javax.validation.Validator validator = validatorFactory.getValidator();
+                final Set<javax.validation.ConstraintViolation<Object>> violations = validator.validate(value);
+                final Locale localeToUse = (context == null || context.getAcceptLanguage() == null) ? Locale.getDefault() : Locale.forLanguageTag(context.getAcceptLanguage().split(",", 2)[0]);
```

Comment:
The perfect way is to use @Inject Lang and then lang.getLanguage(context, Optional.absent())

The language can not only be in the accept header, but also inside a cookie that forces Ninja to use a certain language. Lang supports that.

You can inject Lang into JSRValidator's constructor.